tools/change-registry.sh

```diff
@@ -0,0 +1,143 @@
+            Registry auth user name.
+    --password <PASSWORD>
+            Registry auth user password.
+    --registry-url <SCHEME://REGISTRY_URL/PATH>
+            Registry URL.
+    --ca-file <FILENAME>
+            File containing CA certificate for validating registry self-signed certificate.
+    --help|-h
+            Print this message.
+" "$0"
+}
+
+function parse_args() {
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+    --user)
+      REGISTRY_USER="$2"
+      ;;
+    --password)
+      REGISTRY_PASS="$2"
+      ;;
+    --registry-url)
+      REGISTRY_URL="$2"
+      ;;
+    --ca-file)
+      REGISTRY_CAFILE="$2"
+      ;;
+    --help | -h)
+      usage
+      exit 0
+      ;;
+    --*)
+      echo "Illegal option $1"
+      usage
+      exit 1
+      ;;
+    esac
+    shift $(($# > 0 ? 1 : 0))
+  done
+
+  TEMP_URL="$REGISTRY_URL"
+  REGISTRY_SCHEME="$(grep -oE "^https{0,1}://" <<< "$TEMP_URL")"
+  TEMP_URL="${TEMP_URL#"$REGISTRY_SCHEME"}"
+  REGISTRY_ADDRESS="$(cut -d "/" -f1 <<< "$TEMP_URL")"
+  REGISTRY_PATH="${TEMP_URL#"$REGISTRY_ADDRESS"}"
+  REGISTRY_SCHEME="$(sed "s/:\/\///" <<< "$REGISTRY_SCHEME")"
+  if [[ "$REGISTRY_SCHEME" == "" ]]; then
+    >&2 echo "Cannot parse scheme from registry url: $URL. Scheme should be 'http' or 'https'."
+    exit 1
+  fi
+
+  if [[ "$REGISTRY_ADDRESS" == "" ]]; then
+    >&2 echo "Cannot parse hostname from registry url: $REGISTRY_URL."
+    exit 1
+  fi
+
+  if [[ "$REGISTRY_CAFILE" != "" ]] && [[ ! -f "$REGISTRY_CAFILE" ]]; then
+    >&2 echo "Cannot find ca file: $REGISTRY_CAFILE."
+    exit 1
+  fi
+}
+
+function create_dockerconfigjson() {
+  if [[ "$REGISTRY_USER" != "" ]] && [[ "$REGISTRY_PASS" != "" ]]; then
+    REGISTRY_AUTH="\"auth\":\"$(echo -n "$REGISTRY_USER:$REGISTRY_PASS" | base64 -w0)\""
+  fi
+  cat - <<EOF
+{
+  "auths": {
+    "$REGISTRY_ADDRESS": {
+        $REGISTRY_AUTH
+    }
+  }
+}
+EOF
+}
+
+function create_secret_manifest() {
+  cat - <<EOF
+apiVersion: v1
+type: kubernetes.io/dockerconfigjson
+kind: Secret
+metadata:
+  annotations:
+    helm.sh/resource-policy: keep
+    meta.helm.sh/release-name: deckhouse
+    meta.helm.sh/release-namespace: d8-system
+  labels:
+    app: registry
+    app.kubernetes.io/managed-by: Helm
+    heritage: deckhouse
+    module: deckhouse
+  name: deckhouse-registry
+  namespace: d8-system
+data:
+  .dockerconfigjson: $1
+  address: $2
+  path: $3
+  scheme: $4
+EOF
+  if [[ $5 != "" ]]; then
+    echo "  ca: $5"
+  fi
+}
+
+parse_args "$@"
+
+DOCKERCONFIGJSON="$(create_dockerconfigjson | base64 -w0)"
+REGISTRY_ADDRESS="$(echo -n $REGISTRY_ADDRESS | base64 -w0)"
+REGISTRY_PATH="$(echo -n $REGISTRY_PATH | base64 -w0)"
+REGISTRY_SCHEME="$(echo -n $REGISTRY_SCHEME | base64 -w0)"
+if  [[ "$REGISTRY_CAFILE" != "" ]]; then
+  REGISTRY_CAFILE="$(base64 -w0 < $REGISTRY_CAFILE)"
+fi
+
+create_secret_manifest "$DOCKERCONFIGJSON" "$REGISTRY_ADDRESS" "$REGISTRY_PATH" "$REGISTRY_SCHEME" "$REGISTRY_CAFILE" | kubectl apply -f -
```

Comment:
Looks good. I have a small suggestion. Why not to use `kubectl patch` command to update secret?
```
kubectl patch secret deckhouse-registry -p='{"data":{".dockerconfigjson": "dGVzdAo=", "address": "dGVzdAo=", "path": "dGVzdAo=", "scheme": "dGVzdAo="}}'
```

Reply:
> Looks good. I have a small suggestion. Why not to use `kubectl patch` command to update secret?
> 
> ```
> kubectl patch secret deckhouse-registry -p='{"data":{".dockerconfigjson": "dGVzdAo=", "address": "dGVzdAo=", "path": "dGVzdAo=", "scheme": "dGVzdAo="}}'
> ```

good sugestion, i used it )